provider/pkg/gen/overrides.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2021, Pulumi Corporation.
+
+package gen
+
+import "github.com/pulumi/pulumi-gcp-native/provider/pkg/discovery"
+
+// httpMethodOverride allows overriding the HTTP rest method used to create/get/delete/patch
+// a resource. If set to empty string, the default naming convention is assumed.
+type httpMethodOverride struct {
+	Create string
+	Get    string
+	Delete string
+	Patch  string
+}
+
+func (h httpMethodOverride) retrieve(resource discovery.RestResource, methodName string) *discovery.RestMethod {
+	if m, ok := resource.Methods[methodName]; ok {
+		return &m
+	}
+	return nil
+}
+
+func (h httpMethodOverride) retrieveCreate(resource discovery.RestResource) *discovery.RestMethod {
+	return h.retrieve(resource, h.Create)
+}
+
+func (h httpMethodOverride) retrieveGet(resource discovery.RestResource) *discovery.RestMethod {
+	return h.retrieve(resource, h.Get)
+}
+
+func (h httpMethodOverride) retrieveDelete(resource discovery.RestResource) *discovery.RestMethod {
+	return h.retrieve(resource, h.Delete)
+}
+
+func (h httpMethodOverride) retrievePatch(resource discovery.RestResource) *discovery.RestMethod {
+	return h.retrieve(resource, h.Patch)
+}
+
+// overrides contains a mapping from resource token to http methods which follow non-standard
+// naming patterns.
+var overrides = map[string]httpMethodOverride{
+	"gcp-native:dataproc/v1:RegionJob": {
+		Create: "submit",
```

Comment:
Are there downsides of always taking the `submit` and `register` methods as Create? Are there cases when we don't want to do this?

Reply:
I tried this and it does include a few more resources - e.g. see below metadata diff (in addition to the 4 added in this PR):
```
@@ -3615,6 +3615,22 @@
                 "workerPoolsId"
             ]
         },
+        "gcp-native:cloudchannel/v1:Account": {
+            "baseUrl": "https://cloudchannel.googleapis.com/",
+            "createPath": "v1/accounts/{accountsId}:register",
+            "createParams": [
+                "accountsId"
+            ],
+            "createProperties": {
+                "serviceAccount": {}
+            },
+            "idPath": "v1/accounts/{accountsId}:register",
+            "idParams": [
+                "accountsId"
+            ],
+            "noGet": true,
+            "noDelete": true
+        },
         "gcp-native:cloudchannel/v1:AccountChannelPartnerLink": {
             "baseUrl": "https://cloudchannel.googleapis.com/",
             "createPath": "v1/accounts/{accountsId}/channelPartnerLinks",
@@ -3738,6 +3754,17 @@
             ],
             "noDelete": true
         },
+        "gcp-native:clouddebugger/v2:ControllerDebuggee": {
+            "baseUrl": "https://clouddebugger.googleapis.com/",
+            "createPath": "v2/controller/debuggees/register",
+            "createParams": null,
+            "createProperties": {
+                "debuggee": {}
+            },
+            "idPath": "v2/controller/debuggees/register",
+            "noGet": true,
+            "noDelete": true
+        },
         "gcp-native:cloudfunctions/v1:Function": {
             "baseUrl": "https://cloudfunctions.googleapis.com/",
             "createPath": "v1/projects/{projectsId}/locations/{locationsId}/functions",
@@ -34270,6 +34297,22 @@
             "noGet": true,
             "noDelete": true
         },
+        "gcp-native:webrisk/v1:Uri": {
+            "baseUrl": "https://webrisk.googleapis.com/",
+            "createPath": "v1/projects/{projectsId}/uris:submit",
+            "createParams": [
+                "projectsId"
+            ],
+            "createProperties": {
+                "submission": {}
+            },
+            "idPath": "v1/projects/{projectsId}/uris:submit",
+            "idParams": [
+                "projectsId"
+            ],
+            "noGet": true,
+            "noDelete": true
+        },
         "gcp-native:websecurityscanner/v1:ScanConfig": {
             "baseUrl": "https://websecurityscanner.googleapis.com/",
             "createPath": "v1/projects/{projectsId}/scanConfigs",
```

Looks like these have no delete and no update either. I do notice we have similar resources in the schema currently but wasn't sure what the deletion semantics for these would be? If we are good to add these I am happy to switch away from the overrides and just enable these method names.

Reply:
I have no idea, TBH, would need to find a reference point like a TF resource or something. We could filter those resources out based on the lack of delete/update rather than the name of the method.

On a higher level, while what we have now works for a prototype, maybe it's time to start from scratch, restrict the rules to full CRUD, see what drops out, and then start adding other "patterns" in a consistent manner.